Mathlib/Data/Bool/Basic.lean

```diff
@@ -261,85 +241,46 @@ theorem not_or_self : ∀ x, (!x || x) = true := by decide
 
 theorem bne_eq_xor : bne = xor := by funext a b; revert a b; decide
 
-theorem xor_comm : ∀ a b, xor a b = xor b a := by decide
 #align bool.bxor_comm Bool.xor_comm
 
-@[simp]
-theorem xor_assoc : ∀ a b c, xor (xor a b) c = xor a (xor b c) := by decide
+attribute [simp] xor_assoc
 #align bool.bxor_assoc Bool.xor_assoc
 
-theorem xor_left_comm : ∀ a b c, xor a (xor b c) = xor b (xor a c) := by decide
 #align bool.bxor_left_comm Bool.xor_left_comm
-
-@[simp]
-theorem xor_not_left : ∀ a, xor (!a) a = true := by decide
 #align bool.bxor_bnot_left Bool.xor_not_left
-
-@[simp]
-theorem xor_not_right : ∀ a, xor a (!a) = true := by decide
 #align bool.bxor_bnot_right Bool.xor_not_right
 
-@[simp]
-theorem xor_not_not : ∀ a b, xor (!a) (!b) = xor a b := by decide
+attribute [simp] xor_not_not
 #align bool.bxor_bnot_bnot Bool.xor_not_not
 
-@[simp]
-theorem xor_false_left : ∀ a, xor false a = a := by decide
 #align bool.bxor_ff_left Bool.xor_false_left
-
-@[simp]
-theorem xor_false_right : ∀ a, xor a false = a := by decide
 #align bool.bxor_ff_right Bool.xor_false_right
-
-theorem and_xor_distrib_left (a b c : Bool) : (a && xor b c) = xor (a && b) (a && c) := by
-  cases a <;> simp
 #align bool.band_bxor_distrib_left Bool.and_xor_distrib_left
-
-theorem and_xor_distrib_right (a b c : Bool) : (xor a b && c) = xor (a && c) (b && c) := by
-  cases a <;> cases b <;> cases c <;> simp
 #align bool.band_bxor_distrib_right Bool.and_xor_distrib_right
 
 theorem xor_iff_ne : ∀ {x y : Bool}, xor x y = true ↔ x ≠ y := by decide
 #align bool.bxor_iff_ne Bool.xor_iff_ne
 
 /-! ### De Morgan's laws for booleans-/
 
-@[simp]
-theorem not_and : ∀ a b : Bool, (!(a && b)) = (!a || !b) := by decide
+attribute [simp] not_and
 #align bool.bnot_band Bool.not_and
 
-@[simp]
-theorem not_or : ∀ a b : Bool, (!(a || b)) = (!a && !b) := by decide
+attribute [simp] not_or
 #align bool.bnot_bor Bool.not_or
 
-theorem not_inj : ∀ {a b : Bool}, (!a) = !b → a = b := by decide
 #align bool.bnot_inj Bool.not_inj
 
--- Porting note: having to unfold here is not pretty.
--- There is a discussion on zulip about this at
--- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/LinearOrder.20in.20mathlib3.2F4/near/308228493
 instance linearOrder : LinearOrder Bool where
-  le := fun a b ↦ a = false ∨ b = true
-  le_refl := by unfold LE.le; decide
-  le_trans := by unfold LE.le; decide
-  le_antisymm := by unfold LE.le Preorder.toLE; decide
-  le_total := by unfold LE.le Preorder.toLE PartialOrder.toPreorder; decide
-  decidableLE := by unfold LE.le Preorder.toLE PartialOrder.toPreorder; exact inferInstance
-  decidableEq := inferInstance
-  max := or
-  max_def := λ a b => by cases a <;> cases b <;> decide
-  min := and
-  min_def := λ a b => by cases a <;> cases b <;> decide
```

Comment:
Are these automatic now? My worry is that you end up with the defaults, so I'd prefer to see `min` and `max` filled explicitly.

Reply:
`min` and `max` are set in Std4.

Reply:
Is there a reason you decided to change the defeq in Std? The lemmas in #7984 aren't true by rfl any more.

Reply:
Fixed in std4#329 and #8005